src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -536,6 +535,25 @@ static grpc_byte_buffer* get_serialized_start_client(
                                           upb_StringView_FromString(ptr->data));
     ptr = ptr->next;
   }
+  // This ensures the token string is available when the proto gets serialized.
+  std::optional<std::string> access_token = std::nullopt;
+  // Set access token if the token fetcher is available.
+  grpc::alts::TokenFetcher* token_fetcher =
+      (reinterpret_cast<grpc_alts_credentials_client_options*>(client->options))
+          ->token_fetcher.get();
+  if (token_fetcher != nullptr) {
+    absl::StatusOr<std::string> token = token_fetcher->GetToken();
+    if (!token.ok()) {
+      LOG_EVERY_N(ERROR, 1000) << "Failed to get token from the token fetcher "
+                                  "in client start handshake: "
+                               << token.status();
+    }
+    access_token = *std::move(token);
+  }
+  if (access_token.has_value()) {
+    grpc_gcp_StartClientHandshakeReq_set_access_token(
```

Comment:
nit: we can move this up into the `if (token_fetcher != nullptr)` block

then we can remove `std::optional<std::string> access_token` and directly pass `token`

Reply:
We will still need to hold the token string outside the block though. Because the upb proto only takes a string view, until it's serialized and returned, the original owner of the string object has to stay in scope. So I don't think moving this in the block will save much.

btw, I'll work on addressing the comments later. Just wanted to clarify this point while I'm here.